src/main/java/io/lettuce/core/resource/AddressResolverGroupProvider.java

```diff
@@ -51,10 +50,14 @@
         return ADDRESS_RESOLVER_GROUP;
     }
 
-    private static AddressResolverGroup<?> defaultDnsAddressResolverGroup() {
```

Comment:
Here causes NoClassDefError because there is an upcast `DnsAddressResolverGroup` -> `AddressResolverGroup`) and `DefaultDnsCnameCache` -> `DnsCnameCache`.
I moved these to static class